scripts/check-sample-lnd-conf.sh

```diff
@@ -0,0 +1,136 @@
+# 3. Checks that all default values of lnd are mentioned correctly, including
+#    empty defaults and booleans which are set to false by default.
+
+
+# We are reading the default values of lnd from lnd --help. To avoid formatting
+# issues the width of the terminal is set to 240.
+stty cols 240
+set -e
+
+
+# LND_OPTIONS is a list of all options of lnd including the equal sign, which
+# is needed to distinguish between booleans and other variables.
+# It is created by reading the first two columns of lnd --help. 
+LND_OPTIONS="$(go run -tags="autopilotrpc signrpc walletrpc chainrpc \
+    invoicesrpc watchtowerrpc neutrinorpc monitoring peersrpc kvdb_postgres \
+    kvdb_etcd kvdb_sqlite" github.com/lightningnetwork/lnd/cmd/lnd --help | \
+    awk '{
+        option=""; 
+        if ($1 ~ /^--/){option=$1};
+        if ($2 ~ /^--/){option=$2}; 
+        if (match(option,  /--[^=]+[=]*/))
+            {printf "%s ", substr(option, RSTART, RLENGTH)}
+        } 
+        END { printf "%s", "--end"}')"
+
+
+# LND_HELP is the whole text of lnd --help without any line breaks.
+LND_HELP="$(go run -tags="autopilotrpc signrpc walletrpc chainrpc \
+    invoicesrpc watchtowerrpc neutrinorpc monitoring peersrpc kvdb_postgres \
+    kvdb_etcd kvdb_sqlite" github.com/lightningnetwork/lnd/cmd/lnd --help  | \
+    tr -d '\n')"" --end"
+
+
+# OPTIONS_NO_CONF is a list of all options without any expected entries in 
+# sample-lnd.conf. There's no validation needed for these options. 
+OPTIONS_NO_CONF="help lnddir configfile version litecoin.signet \
+    litecoin.signetchallenge litecoin.signetseednode end"
+
+
+# OPTIONS_NO_LND_DEFAULT is a list of options with default values set, but
+# there aren't any returned defaults by lnd --help. Defaults have to be included 
+# in sample-lnd.conf but no further checks are performed.
+OPTIONS_NO_LND_DEFAULT="bitcoin.dnsseed litecoin.dnsseed"
+
+
+# EXITCODE is returned at the end after all checks are performed and set to 1 
+# if a validation error occurs
+EXITCODE=0
+CONF_FILE=${1:-sample-lnd.conf}
+
+
+for OPTION in $LND_OPTIONS; do
+
+    # Determination of the clean name of the option without leading -- and 
+    # possible = at the end.
+    OPTION_NAME=${OPTION##--}
+    OPTION_NAME=${OPTION_NAME%=}
+
+
+    # Skip if no checks are needed
+    echo "$OPTIONS_NO_CONF" | grep -qw $OPTION_NAME && continue
+
+
+    # Determine the default value of lnd. If the option has no equal sign, 
+    # it is boolean and set to false. 
+    # For other options we grep the text between the current option and the next
+    # option from LND_HELP. The default value is given in brackets (default: xx)
+    # HOME and HOSTNAME are replaced with general values.
+    if [[ "$OPTION" == *"="* ]]; then
+        OPTION_NEXT=$(echo "$LND_OPTIONS" | grep -Po "(?<=$OPTION )[^ ]*")
+        DEFAULT_VALUE_LND="$(echo $LND_HELP | \
+            grep -Po "(?<=\--${OPTION##--}).*(?=\--${OPTION_NEXT##--})" | \
```

Comment:
I wonder if all of these patterns/options are portable across Unixes? I guess we need someone with at least MacOS to test this as well.

Reply:
Good point. What is the easiest way to test it? I am on linux and haven't installed something like a MacOS VM yet. Any idea?

Reply:
I'll just ask for a second reviewer who has a MacOS to test this. In our experience if it works on Linux and Darwin it tends to work on most Unix like systems.

Reply:
Just tested it with MacOS 12.6 but it doesn't know `grep -P`. Ran like this:
```
./scripts/check-sample-lnd-conf.sh sample-lnd.conf                                                                                      ✔  14:17:02 
grep: invalid option -- P
```

Reply:
@hieblmi Thx for check
Is `grep -E` available? Maybe I can adjust the script later.

Reply:
```
 ~ grep --version      
grep (BSD grep, GNU compatible) 2.6.0-FreeBSD
 ~ grep                   
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
```